crates/control/src/kick_selector.rs

```diff
@@ -132,53 +146,209 @@ impl KickSelector {
     }
 }
 
+fn is_ball_in_opponents_corners(
+    ball_position: Point2<Field>,
+    field_dimensions: &FieldDimensions,
+    parameters: &DecisionParameters,
+) -> bool {
+    let left_opponent_corner =
+        field_dimensions.corner(Half::Opponent, field_dimensions::Side::Left);
+    let right_opponent_corner =
+        field_dimensions.corner(Half::Opponent, field_dimensions::Side::Right);
+    let ball_near_left_opponent_corner =
+        distance(ball_position, left_opponent_corner) < parameters.distance_to_corner;
+    let ball_near_right_opponent_corner =
+        distance(ball_position, right_opponent_corner) < parameters.distance_to_corner;
+    ball_near_left_opponent_corner || ball_near_right_opponent_corner
+}
+
+fn determine_playing_situation(
+    filtered_game_controller_state: Option<&FilteredGameControllerState>,
+    ball_position: Point2<Field>,
+    field_dimensions: &FieldDimensions,
+    parameters: &DecisionParameters,
+) -> PlayingSituation {
+    let is_ball_in_opponent_corner =
+        is_ball_in_opponents_corners(ball_position, field_dimensions, parameters);
+    match filtered_game_controller_state {
+        Some(FilteredGameControllerState {
+            game_state: FilteredGameState::Playing { kick_off: true, .. },
+            game_phase: GamePhase::Normal,
+            opponent_game_state:
+                FilteredGameState::Playing {
+                    ball_is_free: false,
+                    ..
+                },
+            ..
+        }) => PlayingSituation::KickOff,
+        Some(FilteredGameControllerState {
+            game_phase: GamePhase::PenaltyShootout { .. },
+            kicking_team: Team::Hulks,
+            ..
+        })
+        | Some(FilteredGameControllerState {
+            sub_state: Some(SubState::PenaltyKick),
+            kicking_team: Team::Hulks,
+            ..
+        }) => PlayingSituation::PenaltyShot,
+        _ if is_ball_in_opponent_corner => PlayingSituation::CornerKick,
```

Comment:
I would limit this to `FilteredGameState::Playing` to reduce the probability of errors

Reply:
There is no way these decisions can be used outside `Playing`. There is no ball and there is no striker...

Reply:
Ah sorry, I think I meant `GamePhase`